### DISCUSSION TITLE
Pixi vs UPM?

### DISCUSSION THREAD

**Discussion by AYA-123** [prefix-dev/pixi#787] [Q&A]
thoughts on https://github.com/replit/upm vs pixi? Are there advantages or disadvatages to either?

**Comment by ruben-arts** [comment]
He @AYA-123, I haven't used it but reading from their readme there is a significant difference.

> UPM does not implement package management itself. Instead, it runs a package manager for you. 

Whereas pixi is an actual package manager. We don't just try to install as many ecosystems as possible but rather allow you to install more things through the conda package ecosystem. Our approach to how we solve and install sets us apart.

Because the conda package is language and platform agnostic you can use it for anything. Which does make it feel like a Universal Package Manager. But not in the same sense as UPM is.

It does seem like a cool project I should try out though!

**Comment by AYA-123** [comment]
Thanks for the reply! So UPM just helps organise/suggest packages - not
actually install them, while pixi can install packages from basically
anywhere?

**Reply by ruben-arts** [reply]
Ah no I think that is the wrong way of looking at it. UPM manages package managers, so possibly they would want to also support `pixi` for the conda ecosystem for instance. In their example you can see the way it works:
```shell
$ upm -l python add flask
--> python3 -m poetry init --no-interaction
...

--> python3 -m poetry add flask
Creating virtualenv python-py3.7 in /root/.cache/pypoetry/virtualenvs
Using version ^1.1 for flask
```
Here `upm` runs `poetry`, which is also a package manager for python. It can orchestrate these other package managers and understand their way of working so the can add a universal layer to it.

**Comment by AYA-123** [comment]
Cool! Thanks
